src/fsharp/FSharp.Core/fslib-extra-pervasives.fs

```diff
@@ -70,6 +81,78 @@ module ExtraTopLevelOperators =
                     let key = RuntimeHelpers.StructBox(k)
                     if d.ContainsKey(key) then (r <- d.[key]; true) else false
                 member s.Remove(k : 'Key) = (raise (NotSupportedException(SR.GetString(SR.thisValueCannotBeMutated))) : bool) 
+#if FX_NO_READONLY_COLLECTIONS
```

Comment:
if FX_ATLEAST_45 && not FSHARP_CORE_PORTABLE

but why not explicitly exlude with ne flag. seems like a common pattern here